mailpoet/lib/Newsletter/Scheduler/PostNotificationScheduler.php

```diff
@@ -95,7 +108,11 @@ public function schedulePostNotification($postId) {
     }
   }
 
-  public function createPostNotificationSendingTask(NewsletterEntity $newsletter): ?SendingTask {
+  /**
+   * @param NewsletterEntity $newsletter
+   * @return SendingQueueEntity|null
+   */
+  public function createPostNotificationSendingTask(NewsletterEntity $newsletter): ?SendingQueueEntity {
```

Comment:
This method has name `createPostNotificationSendingTask` and you return `SendingQueueEntity`. I would expect to return `ScheduledTaskEntity` instead. Or is there a reason for this? I checked it, and we use the returned value only in tests.

Reply:
The reason is that we have `$sendingQueue->getTask()` but not a method to get the queue from `ScheduledTaskEntity` and the returned values are only used in test as you mentioned, still I'd agree the name doesn't match much, will adapt

Reply:
@samnajian It's possible to change the unidirectional relation to bidirectional. I did it in a different PR in this commit https://github.com/mailpoet/mailpoet/pull/4305/commits/33fe72bb5d1dcef0a1eb9ed1167cb6fc3543a09b.
You can read how it works [here](https://www.doctrine-project.org/projects/doctrine-orm/en/latest/reference/association-mapping.html#one-to-one-bidirectional).

You changed the method's name in the newest version, but other schedulers have the suffix `SendingTask`. So I think that we should preserve this pattern.

Reply:
Since you've added the code needed here in [this PR](https://github.com/mailpoet/mailpoet/pull/4305) it makes sense to either wait until the mentioned PR is merged or based [PR 4314](https://github.com/mailpoet/mailpoet/pull/4314) on it, what do you think?

Reply:
@samnajian Yes, it makes sense.